### DISCUSSION TITLE
Can I run dagger with docker in docker and use that to support CI environments that don't allow privileged access?

### DISCUSSION THREAD

**Discussion by red8888** [dagger/dagger#12468] [Help]
### What happened? What did you expect to happen?

This might be a dumb question but can I run this in the official docker in docker image to get around my perms issues?

Bitbucket pipelines don't support enough privilege for dagger to work: https://github.com/dagger/dagger/discussions/1677#discussioncomment-2619008. And I'm not going to run them on-prem which is the workaround.

But could I have my dagger scripts built into the official docker in docker image (https://hub.docker.com/_/docker) and then in my bb pipeline use that image to do the builds?

Does it not matter and will it still need host level perms that bb would reject?

**Comment by codethief** [comment]
I'd also be very much interested in this question! Unfortunately, it seems [the answer so far is negative](https://docs.dagger.io/configuration/engine/#security): 

> "Rootless mode" means running the Dagger Engine as a container without the --privileged flag. In this case, the container would not run as the root user of the system.

> Currently, the Dagger Engine cannot be run as a rootless container; network and filesystem constraints related to rootless usage would currently significantly limit its capabilities and performance.

This matches my experience with nesting containers in general: Without privileges it's nigh impossible, especially in corporate environments where you also face constraints by AppArmor or seccomp.

**Comment by codethief** [comment]
Related ticket: https://github.com/dagger/dagger/issues/3140
